cbr-service-profile/main.tf

```diff
@@ -0,0 +1,108 @@
+    name             = "${var.prefix}-cbr-vpc-zone"
+    account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+    zone_description = "cbr-vpc-zone-terraform"
+    addresses = [
+      for zone_vpc_crn in var.zone_vpc_crn_list :
+      { "type" = "vpc", value = zone_vpc_crn }
+    ]
+  }] : []
+
+  service_ref_zone_list = (length(var.zone_service_ref_list) > 0) ? [{
+    name             = "${var.prefix}-cbr-serviceref-zone"
+    account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+    zone_description = "cbr-serviceref-zone-terraform"
+    # when the target service is containers-kubernetes or any icd services, context cannot have a serviceref
+    addresses = [
+      for serviceref in var.zone_service_ref_list : {
+        type = "serviceRef"
+        ref = {
+          account_id   = data.ibm_iam_account_settings.iam_account_settings.account_id
+          service_name = serviceref
+        }
+      }
+    ]
+  }] : []
+  zone_list = concat(tolist(local.vpc_zone_list), tolist(local.service_ref_zone_list))
+}
+
+module "cbr_zone" {
+  count            = length(local.zone_list)
+  source           = "../cbr-zone-module"
+  name             = local.zone_list[count.index].name
+  zone_description = local.zone_list[count.index].zone_description
+  account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+  addresses        = local.zone_list[count.index].addresses
+}
+
+locals {
+  rule_contexts = [{
+    attributes = [
+      {
+        "name" : "endpointType",
+        "value" : "private"
+      },
+      {
+        name  = "networkZoneId"
+        value = join(",", ([for zone in module.cbr_zone : zone.zone_id]))
+    }]
+  }]
+}
+
+module "cbr_rule" {
+  count            = length(var.target_service_details)
+  source           = "../cbr-rule-module"
+  rule_description = "${var.prefix}-${var.target_service_details[count.index].target_service_name}-serviceprofile-rule"
+  enforcement_mode = var.target_service_details[count.index].enforcement_mode
+  rule_contexts    = local.rule_contexts
+  operations = (length(lookup(local.operations_apitype_val, var.target_service_details[count.index].target_service_name, [])) > 0) ? [{
+    api_types = [
+      # lookup the map for the target service name, if not present make api_type_id as empty
+      for apitype in lookup(local.operations_apitype_val, var.target_service_details[count.index].target_service_name, []) : {
+        api_type_id = apitype
+    }]
+  }] : []
+
+  resources = [{
+    tags = var.target_service_details[count.index].tags
+    attributes = [
+      {
+        name : "accountId",
+        operator = "stringEquals",
+        value : data.ibm_iam_account_settings.iam_account_settings.account_id
+      },
+      {
+        name     = "serviceName",
+        operator = "stringEquals",
+        value    = var.target_service_details[count.index].target_service_name
+    }]
+  }]
```

Comment:
we probably want to add the option to append extra attributes here. I'm thinking if we wanted to restrict to a resource group, or instance (although I don't think restricting by instance is a use case for this?)

Reply:
Yes.  Service instance is not a usecase here.  So adding only the resource group id.